src/regdesk/states/multiple_result_state.js

```diff
@@ -0,0 +1,55 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * Multiple result state
+ */
+export class MultipleResultState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
+    return {
+      CANCEL: 'CANCEL',
+      SELECTED_SINGLE_RESULT: 'SELECTED_SINGLE_RESULT',
+    };
+  }
+
+  /**
+   * Initializes a new instance of the MultipleResultState class.
+   * @param {RegMachineArgs} regMachineArgs - Standard argument set for states.
+   * @param {*} eventMap - Mapping of this object's transition events to new states.
+   */
+  constructor(regMachineArgs, eventMap) {
+    super(regMachineArgs, eventMap);
+
+    this.button1 = this.screenRow.querySelector('#multipleResultExampleOne');
+    this.button1.addEventListener('click', () => {
+      this.dispatchTransition(
+        MultipleResultState.events.SELECTED_SINGLE_RESULT,
+        { badgeLine1: 'ASDF', fromMultiple: true });
```

Comment:
The "fromMultiple" should not be part of the response details. Mostly because it's the job of the mgr to know the flow of states and this is kind of going around that. I would expect mgr to be smart enough to know a cancel from here leads back to multiple vs single. Otherwise you're kind of blurring the boundaries between the responsibilities of a state (present information and buttons) and the mgr (handle responses to those buttons ie state transitions).

Reply:
I agree, this was expedient but rubbed me the wrong way. I'm expecting to hoist most of this state information out into the commMgr (which doesn't exist yet) and have very limited info, if anything, dropped into the event object.